### PR TITLE
Feauture/enum

### PR DESCRIPTION
**Pull request by whaeck**
Lifted the enumeration code out of the feature/v1.9 branch

### CODE REVIEW

File: src/utility/enumeration.hpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+ *  @return An enumeration value derived from the string symbol
+ */
+template < typename Enumeration,
+           typename = typename std::enable_if< std::is_enum< Enumeration >::value >::type >
+const Enumeration& fromString( const std::string& symbol ) {
+
+  auto found = Map< Enumeration >::symbols.find( symbol );
+  if ( found == Map< Enumeration >::symbols.end() ) {
+
+    Log::error( "An enumeration symbol for {} has no registered value: \'{}\'",
```
Comment (staleyLANL) [inline src/utility/enumeration.hpp:70]: Perhaps escaped double-quotes here, not single quotes, as one typically thinks of strings (as opposed to single characters) as being in double quotes.

**Review by staleyLANL — COMMENTED** [review]
Just a handful of minor comments. Looks good.

File: src/GNDStk/enums/GridStyle.hpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+} // enums namespace
+} // GNDStk namespace
+} // njoy namespace
+
+namespace njoy {
+namespace utility {
+namespace enumeration {
+
+  /**
+   *  @brief Template specialisation to convert Length to/from strings
```
[the remark this answers is not in the document]
Comment (whaeck) [reply src/GNDStk/enums/GridStyle.hpp:53]: A copy/paste error :-)

File: src/GNDStk/enums/Interpolation.hpp [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+namespace GNDStk {
+namespace enums {
+
+  /**
+   *  @class
+   *  @brief Enumeration class giving acceptable interpolation schemes
+   *
+   *  See GNDS v1.9 specifications section 3.5.3
+   */
+  enum class Interpolation {
```
[the remark this answers is not in the document]
Comment (whaeck) [reply src/GNDStk/enums/Interpolation.hpp:19]: Yeah, that's why I was like "I'll lift that code out of the 1.9 proto branch".

File: src/utility/enumeration.hpp (hunk 1 above)
Comment (whaeck) [reply src/utility/enumeration.hpp]: Consider it changed.

**Comment by staleyLANL** [comment]
Oops, looks like some of the Continuous Integration stuff failed.

**Comment by whaeck** [comment]
Strange. The actions themselves indicate a full success. I relaunched all tests and they're all passing now. Must have been an issue on github's side. Anyway, all tests should have passed in the next half hour or so.

**Review by staleyLANL — APPROVED** [review]
Approved.

**Pull request merged**